### PR TITLE
SimpleHorizontalSingleSelect

### PR DESCRIPTION
**Pull request by goranalkovic-infinum**
# Description

Sometimes `SimpleVerticalSingleSelect` takes up too much space. That's where `SimpleHorizontalSingleSelect` comes in!

It offers you a nice way of picking between 2 and 5 options (you can do more of course, no limits! 😄 )

Highlighted features:
- 3 styles (borderless / inset border / offset border - see image below)
- option to separately define tooltip for each button
- option to provide button labels/icons/tooltips separately from values
- multiple align options
- replace `label` with `inlineLabel` if you want the control and the label on the same level without messing with extra divs 😄  (if you put an `IconLabel` inside, make sure it has `standalone` prop set)

# Screenshots / Videos

image

# Linked documentation PR

\-

### CODE REVIEW

File: scripts/components/simple-horizontal-single-select/docs/readme.mdx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# SimpleHorizontalSingleSelect
+
+A simple component for picking mutually exclusive options.
+
+## Usage
+
+```jsx
+<SimpleHorizontalSingleSelect
+	label='Size'
```
Comment (volfkarlo) [inline scripts/components/simple-horizontal-single-select/docs/readme.mdx:9]: Maybe to add translation just to encourage to do so.

File: scripts/components/simple-horizontal-single-select/docs/readme.mdx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+/>
+```
+
+Each option should be in this format:
+
+```js
+{
+	icon: icons.large, // Optional, can be either ES icon (name or whole icon element) or dashicon name.
+	value: 'l',
+	label: 'L', // Optional
```
Comment (volfkarlo) [inline scripts/components/simple-horizontal-single-select/docs/readme.mdx:22]: Same here 😄

File: scripts/components/simple-horizontal-single-select/docs/story.js [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+import { icons } from '@eightshift/frontend-libs/scripts';
+import readme from '@eightshift/frontend-libs/scripts/components/simple-horizontal-single-select/docs/readme.mdx';
+import { SimpleHorizontalSingleSelect } from '../simple-horizontal-single-select';
+import { ucfirst } from '../../../editor/ucfirst';
+
+export default {
+	title: 'Options/SimpleHorizontalSingleSelect',
+	parameters: {
+		docs: {
+			page: readme
```
Comment (volfkarlo) [inline scripts/components/simple-horizontal-single-select/docs/story.js:11]:
```suggestion
			page: readme,
```

File: scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+ * @param {boolean} [props.iconOnly=false]                                          - If `true`, the buttons will only contain icons. If a label is also passed, it will be used for the button tooltip.
+ * @param {boolean} [props.allowWrap=true]                                          - If `true` and there is more options then can fit, the buttons will wrap to the row below.
+ * @param {boolean} [props.disabled=false]                                          - If `true`, the component will be disabled.
+ * @param {'none'|'inset'|'offset'} [props.border='none']                           - Sets the appearance of a border around the buttons.
+ * @param {'default'|'stretch'|'left'|'center'|'right'} [props.alignment='default'] - If `true` and there is more options then can fit, the buttons will wrap to the row below.
+ * @param {boolean} [props.includeWpBottomSpacing=false]                            - If `true`, the component will add bottom spacing to match other Gutenberg components.
+ * @param {boolean} [props.compactButtons=false]                                     - If `true`, the buttons are rendered smaller
```
Comment (volfkarlo) [inline scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js:22]: Missed one space 😂

File: scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+		spacingConfig['marginLeft'] = 'auto';
+		spacingConfig['marginRight'] = 'auto';
+		spacingConfig['justifyContent'] = 'center';
+	} else if (alignment === 'right') {
+		spacingConfig['marginLeft'] = 'auto';
+		spacingConfig['justifyContent'] = 'flex-end';
+	} else if (alignment === 'stretch') {
+		spacingConfig['justifyContent'] = 'space-between';
+		spacingConfig['width'] = '100%';
+	}
```
Comment (volfkarlo) [inline scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js:59]: Just a suggestion to pass it as a config?

something like

```
SPACING_CONFG = {
left: {
marginRight: 'auto',
.
.
.
}
}
```

File: scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+	if (border === 'offset') {
+		spacingConfig['borderRadius'] = 3;
+		spacingConfig['padding'] = '0.125rem';
+		spacingConfig['border'] = '1px solid #ccc';
+	}
```
Comment (volfkarlo) [inline scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js:65]: same for this as an enumeration or an object (config) 😄

**Review by volfkarlo — CHANGES_REQUESTED** [review]
Awesome work!

Just a few tiny suggestions! 😄

**Review by iobrado — APPROVED** [review]

**Review by simun-ivanac — APPROVED** [review]

File: scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+			style={{
+				display: 'flex',
+				flexWrap: allowWrap ? 'wrap' : 'nowrap',
+				maxWidth: '100%',
+				...(spacingConfig[alignment] ?? {}),
+				...(borderConfig[border] ?? {}),
+			}}
+		>
+			{options.map((item, i) => {
+				const icon = icons?.[item?.icon] ?? item?.icon ?? optionLabels?.[i]?.icon;
```
Comment (dingo-d) [inline scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js:91]: Stuff of nightmares, but it's in one line 😂 🙈

File: scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+				const label = item?.label ?? optionLabels?.[i]?.label;
+				const current = optionLabels ? item : item?.value;
+				const tooltip = item?.tooltip ?? optionLabels?.[i]?.tooltip ?? item?.label ?? optionLabels?.[i]?.label;
```
Comment (dingo-d) [inline scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js:94]: wtf

**Review by dingo-d — APPROVED** [review]

File: scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js (hunk 8 above)
Comment (goranalkovic-infinum) [reply scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js:94]: Let me break this down
```js
const icon = icons?.[item?.icon] ?? item?.icon ?? optionLabels?.[i]?.icon;
```
`icons?.[item?.icon]` - if string icon name is passed on, try it first with ES icons
`item?.icon` - if no such icon exists, try with just that element (or if string a dashicon)
`optionLabels?.[i]?.icon` - fallback if `optionLabels` is used to set the labels

Comment (dingo-d) [reply scripts/components/simple-horizontal-single-select/simple-horizontal-single-select.js:94]: I know it looks worse, that's why I approved the PR 😂. Still, the one-liner is super hard to read 🙈

**Pull request merged**